### ISSUE TITLE
Extracting learning rate from nnx.optimizer directly

### ISSUE THREAD

**Issue by noahzhy** [google/flax#4320]
How to get learning rate from nnx.optimizer in the training loop?
```python3
for epoch in range(1, num_epochs + 1):
    pbar = tqdm(train_ds)
    for batch in pbar:
        loss_dict = train_step(model, optimizer, batch, loss_fn, epoch)
        # print(lr) How to get lr here?
        pbar.set_description(f'Epoch {epoch:3d}, lr: {lr:.7f}, loss: {loss_dict["loss"]:.4f}')
```

**Comment by IvyZX** [comment]
Optax's [`optax.GradientTransformation`](https://optax.readthedocs.io/en/latest/api/transformations.html#optax.GradientTransformation) only outputs two pure functions for optimizer states and doesn't keep the original hyperparameters like learning rate. It seems that you need to record the learning rate when you create that Optax optimizer instance (like `optax.adam(...)`).

**Comment by noahzhy** [comment]
Thanks for your tips. I figure it out.
```python3
tx = optax.inject_hyperparams(optax.nadam)(lr_fn)
optimizer = nnx.Optimizer(model, tx)

for epoch in range(1, num_epochs + 1):
    pbar = tqdm(train_ds)
    for batch in pbar:
        loss_dict = train_step(model, optimizer, batch, loss_fn, epoch)
        lr = optimizer.opt_state.hyperparams['learning_rate'].value
        pbar.set_description(f'Epoch {epoch:3d}, lr: {lr:.7f}, loss: {loss_dict["loss"]:.4f}')

```
